src/field_theory/adjoin.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2020 Thomas Browning and Patrick Lutz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning and Patrick Lutz
+-/
+
+import field_theory.subfield
+import field_theory.separable
+import field_theory.tower
+import group_theory.subgroup
+import field_theory.minimal_polynomial
+import linear_algebra.dimension
+import linear_algebra.finite_dimensional
+import ring_theory.adjoin_root
+import data.zmod.basic
+
+/-!
+# Adjoining Elements to Fields
+
+In this file we introduce the notion of adjoining elements to fields.
+This isn't quite the same as adjoining elements to rings, since the element might not be algebraic.
+
+## Main results
+
+(This is just a start, We've got more to add, including a proof of the Primitive Element Theorem)
+
+- `adjoin_twice`: adjoining S and then T is the same as adjoining S ∪ T.
+
+## Notation
+
+ - `F[α]` : Adjoin α to F.
+-/
+
+variables (F : Type*) [field F] {E : Type*} [field E] [algebra F E] (S : set E)
+
+/--
+Adjoining a set S ⊆ E to a field F
+-/
+def adjoin : set E := field.closure (set.range (algebra_map F E) ∪ S)
```

Comment:
To clarify, do you mean that we should make a PR with changes to `field_theory/subfield.lean` to change the definition of `field.closure` so that it is defined to be a `subalgebra`?

Reply:
The problem is that field.closure doesn't know that the set contains a field, so you can't know ahead of time that field.closure is a subalgebra.

Reply:
Ok, I've made adjoin automatically a subalgebra. The one downside that I noticed is that I had to put (... : set E) in a few places because lean sometimes had trouble figuring out the case from a subalgebra to a subset.